examples/templates/do-linux/main.tf

```diff
@@ -45,43 +51,179 @@ variable "step2_do_admin_ssh_key" {
+  option {
+    name  = "Debian 10"
+    value = "debian-10-x64"
+    icon  = "/icon/debian.svg"
+  }
+  option {
+    name  = "CentOS Stream 9"
+    value = "centos-stream-9-x64"
+    icon  = "/icon/centos.svg"
+  }
+  option {
+    name  = "CentOS Stream 8"
+    value = "centos-stream-8-x64"
+    icon  = "/icon/centos.svg"
+  }
+  option {
+    name  = "Rocky Linux 8"
+    value = "rockylinux-8-x64"
+    icon  = "/icon/rockylinux.svg"
+  }
+  option {
+    name  = "Rocky Linux 8.4"
+    value = "rockylinux-8-4-x64"
+    icon  = "/icon/rockylinux.svg"
   }
 }
 
-variable "droplet_size" {
-  type        = string
-  description = "Which Droplet configuration would you like to use?"
-  default     = "s-1vcpu-1gb"
-  validation {
-    condition     = contains(["s-1vcpu-1gb", "s-1vcpu-2gb", "s-2vcpu-2gb", "s-2vcpu-4gb", "s-4vcpu-8gb", "s-8vcpu-16gb"], var.droplet_size)
-    error_message = "Value must be s-1vcpu-1gb, s-1vcpu-2gb, s-2vcpu-2gb, s-2vcpu-4gb, s-4vcpu-8gb or s-8vcpu-16gb."
+data "coder_parameter" "droplet_size" {
+  name    = "Which Droplet configuration would you like to use?"
+  default = "s-1vcpu-1gb"
+  type    = "string"
+  icon    = "/icon/memory.svg"
+  mutable = false
+  option {
+    name  = "1 vCPU, 1 GB RAM"
+    value = "s-1vcpu-1gb"
+  }
+  option {
+    name  = "1 vCPU, 2 GB RAM"
+    value = "s-1vcpu-2gb"
+  }
+  option {
+    name  = "2 vCPU, 2 GB RAM"
+    value = "s-2vcpu-2gb"
+  }
+  option {
+    name  = "2 vCPU, 4 GB RAM"
+    value = "s-2vcpu-4gb"
+  }
+  option {
+    name  = "4 vCPU, 8 GB RAM"
+    value = "s-4vcpu-8gb"
+  }
+  option {
+    name  = "8 vCPU, 16 GB RAM"
+    value = "s-8vcpu-16gb"
   }
 }
 
-variable "home_volume_size" {
-  type        = number
-  description = "How large would you like your home volume to be (in GB)?"
-  default     = 20
+
+data "coder_parameter" "home_volume_size" {
+  name        = "How large would you like your home volume to be (in GB)?"
+  description = "This volume will be mounted to /home/coder."
+  type        = "number"
+  default     = "20"
+  mutable     = false
   validation {
-    condition     = var.home_volume_size >= 1
-    error_message = "Value must be greater than or equal to 1."
+    min = 1
+    max = 999999
   }
 }
 
-variable "region" {
-  type        = string
-  description = "Which region would you like to use?"
+data "coder_parameter" "region" {
+  name        = "Which region would you like to use?"
+  description = "This is the region where your workspace will be created."
+  icon        = "/emojis/1f30e.png"
+  type        = "string"
   default     = "ams3"
-  validation {
-    condition     = contains(["nyc1", "nyc2", "nyc3", "sfo1", "sfo2", "sfo3", "ams2", "ams3", "sgp1", "lon1", "fra1", "tor1", "blr1"], var.region)
-    error_message = "Value must be nyc1, nyc2, nyc3, sfo1, sfo2, sfo3, ams2, ams3, sgp1, lon1, fra1, tor1 or blr1."
+  mutable     = false
+  option {
+    name  = "New York 1"
+    value = "nyc1"
+    icon  = "/emojis/1f1fa_1f1f8.png"
```

Comment:
Are those icons available? Shouldn't we upload them?

<img width="553" alt="Screenshot 2023-03-14 at 17 34 31" src="https://user-images.githubusercontent.com/14044910/225074348-b1a972ce-c250-42db-9545-1011cc4f7039.png">

Reply:
This is a typo. I am sorry. It should be `icon  = "/emojis/1f1fa-1f1f8.png"` instead of `icon  = "/emojis/1f1fa_1f1f8.png"`